app/models/elastic_search.rb

```diff
@@ -0,0 +1,157 @@
+# frozen_string_literal: true
+
+class ElasticSearch
+  def initialize(_index = nil)
+    @es_config = MlnConfigurationController.new.elasticsearch_config('teachersets')
+    arguments = {
+      host: @es_config['host'],
+      transport_options: {
+        request: { open_timeout: @es_config['connect_timeout'] },
+        headers: { content_type: 'application/json' }
+      }
+    }
+    @client = Elasticsearch::Client.new(arguments)
+    @current_file = File.basename(__FILE__)
+    @index = @es_config['index'] || 'teacherset'
+    @type = @es_config['type'] || 'teacherset'
+  end
+
+
+  def index_doc_count
```

Comment:
Let's put docs on all of these.

Also the question of unit tests here.  Is this something that can be unit or functional tested?